internal/dataplane/kong_client_test.go

```diff
@@ -595,10 +595,10 @@ func TestKongClientUpdate_ConfigStatusIsNotified(t *testing.T) {
 			)
 
 			kongClient.SetConfigStatusNotifier(statusQueue)
-			for range tc.gatewayFailuresCount {
+			for i := 0; i < tc.gatewayFailuresCount; i++ {
```

Comment:
From Golang 1.22 doing something like that is perfectly fine, [see](https://go.dev/play/p/2Ur3ik4Kkng)

Reply:
For context: this was added in 1.22 https://tip.golang.org/doc/go1.22#language